File: backend/src/main/java/com/becasfind/api/services/FavoritoService.java
package com.becasfind.api.services;

import com.becasfind.api.models.dtos.BecaDTO;

import java.util.List;

public interface FavoritoService {

    void guardar(String email, Long idBeca);

    void eliminar(String email, Long idBeca);

    List<BecaDTO> listar(String email);

    boolean isFavorito(String email, Long idBeca);
}

File: frontend/src/services/favoritoService.ts
import api from './api';
import type { ApiResponse, BecaSummary } from '../types';

export const favoritoService = {
  listar() {
    return api.get<ApiResponse<BecaSummary[]>>('/favoritos');
  },

  guardar(idBeca: number) {
    return api.post<ApiResponse<void>>(`/favoritos/${idBeca}`);
  },

  eliminar(idBeca: number) {
    return api.delete<ApiResponse<void>>(`/favoritos/${idBeca}`);
  },

  check(idBeca: number) {
    return api.get<ApiResponse<{ favorito: boolean }>>(`/favoritos/${idBeca}/check`);
  },
};
